### PR TITLE
Turn on linspace and arange tests.

### PR DESCRIPTION
**Pull request by cowlicks**
PR #115 with the tests broken so nothing was actually tested. This turns the tests on and fixes bugs that the tests uncovered. However there are issues with `dask.array.arange` matching `numpy.arange` in edge cases that are not specified or supported by NumPy. Tests where these edge cases come up were marked as expected failures.

The edge cases come up when arange is passed a floating point `step` argument. The step argument is added repeatedly, which might result in points near the end of the interval being included when they should not.

### CODE REVIEW

File: dask/array/tests/test_creation.py [hunk 1]
```diff
@@ -1,26 +1,18 @@
...
-    if isinstance(c, np.ndarray):
-        c = c.all()
-    return c
+    """a is a dask array, b is a numpy array. Checks dtype, shape and value. No
+    `assert` needed.
+    """
+    a = np.array(a)
+    assert a.shape == b.shape
+    assert a.dtype == b.dtype
+    assert_array_almost_equal(a, b)
```
Comment (mrocklin) [inline dask/array/tests/test_creation.py]: Perhaps the eventual merged `eq` function should behave like this instead of returning True/False

**Comment by mrocklin** [comment]
Seems fine to me.  Merging soon if no comment.
